proposals/function-pointers.md

```diff
@@ -263,62 +272,41 @@ We use `System.Runtime.InteropServices.OutAttribute`, applied as a `modreq` to t
-}
-
-```
-
-Additionally the language would likely also want to:
-
-- Flag any managed calls to a method tagged with `NativeCallable` as an error. Given the function can't be invoked from
-managed code the compiler should prevent developers from attempting such an invocation.
-- Prevent method group conversions to `delegate` when the method is tagged with `NativeCallable`.
-
-This is not necessary to support `NativeCallable` though. The compiler can support the `NativeCallable` attribute as is
-using the existing syntax. The program would simply need to cast to `void*` before casting to the correct `delegate*`
-signature. That would be no worse than the support today.
-
-``` csharp
-void* v = &CloseHandle;
-delegate* cdecl<IntPtr, bool> f1 = (delegate* cdecl<IntPtr, bool>)v;
-```
-
-### Extensible set of unmanaged calling conventions
-
-The set of unmanaged calling conventions supported by the current ECMA-335 encodings is outdated. We have seen requests to add support
-for more unmanaged calling conventions, for example:
-
-- [vectorcall](https://docs.microsoft.com/cpp/cpp/vectorcall) https://github.com/dotnet/coreclr/issues/12120
-- StdCall with explicit this https://github.com/dotnet/coreclr/pull/23974#issuecomment-482991750
-
-The design of this feature should allow extending the set of unmanaged calling conventions as needed in future. The problems include
-limited space for encoding calling conventions (12 out of 16 values are taken in `IMAGE_CEE_CS_CALLCONV_MASK`) and number of places
-that need to be touched in order to add a new calling convention. A potential solution is to introduce a new encoding that represents
-the calling convention using [`System.Runtime.InteropServices.CallingConvention`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.callingconvention) enum.
-
-For reference, https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/IR/CallingConv.h has the list of calling conventions
-supported by LLVM. While it is unlikely that .NET will ever need to support all of them, it demonstrates that the space of calling
-conventions is very rich.
+### Metadata Representation of Calling Conventions
+
+The following calling convention specifiers are specially recognized by the compiler:
+
+* `System.Runtime.CompilerServices.CallConvCdecl`
+* `System.Runtime.CompilerServices.CallConvStdcall`
+* `System.Runtime.CompilerServices.CallConvFastcall`
+* `System.Runtime.CompilerServices.CallConvThiscall`
+
+If the identifier specified for the `unmanaged` calling convention binds to one of these types, then the function pointer
+is emitted with the calling convention bit set to the appropriate value, and encodes the calling convention type as the first
+modopt on the return type of the function pointer. For any other `CallConv` type or combination of such types, the compiler
+emits a calling convention bit of Unmanaged (0x9), and encodes all `CallConv` types specified as the first modopts on the return
```

Comment:
>as the first modopts on the return [](start = 94, length = 34)

Same comment here. And we shouldn't specify any relative order for modifiers because in case of overriding/implementing the syntax doesn't really define the order.